### DISCUSSION TITLE
Unable to fetch the node name & pod name

### DISCUSSION THREAD

**Discussion by AmitJuneja25** [jupyter/enterprise_gateway#1193] [Q&A]
Is there  a way to fetch the node name & pod name for the kernel in GKE (Google Kubernetes Engine)

**Comment by kevin-bates** [comment]
Hi @AmitJuneja25 - thanks for opening this discussion.

We don't provide a mechanism for accessing this kind of information.  Could you please elaborate on your use-case?  _What_ would be fetching the node and pod names for the kernel?  Is this information needed within the kernel's runtime itself (i.e., from within the notebook)?  Are you building some kind of admin application that needs this information?

We do, however, decorate all kernel pods with a [set of labels](https://github.com/jupyter-server/enterprise_gateway/blob/746f8b340dc170b2ec2638b0d146dee8b39d1f1b/etc/kernel-launchers/kubernetes/scripts/kernel-pod.yaml.j2#L15-L19) which can be used to search across the k8s cluster and retrieve this information.  The [`KubernetesProcessProxy` uses these labels](https://github.com/jupyter-server/enterprise_gateway/blob/746f8b340dc170b2ec2638b0d146dee8b39d1f1b/enterprise_gateway/services/processproxies/k8s.py#L75) (namely `kernel_id`) to discover the kernel pod for example.

**Reply by AmitJuneja25** [reply]
Hello @kevin-bates we are using a dashboard application & the node & pod names would be fetched by our backend service which will make a call to the {EG_URL/kid} where EG_URL = enterprise-gateway-url & kid = kernel id. We are expecting some API call which would provide us with the details of the pod name or node name to which the kernel id refers to. We are using EG as a service which is deployed in our GKE cluster.

**Reply by kevin-bates** [reply]
> We are expecting some API call which would provide us with the details of the pod name or node name to which the kernel id refers to.

I understand.  Nothing like this exists and will be non-trivial per [my response on the other discussion item](https://github.com/jupyter-server/enterprise_gateway/discussions/1198#discussioncomment-4148512).  As I noted, this kind of functionality would need to be performed by the provisioner that would need to be "plumbed" and returned to the user.  Since EG is "resource manager agnostic" different "queries" would return different items (e.g., kernels running in Hadoop YARN clusters would return the Application ID) and how those results are expressed is non-trivial.  

Seems like a "status" method would be useful that returns JSON and could be part of the `/api/kernels/<kernel_id>` `GET` response - although we'd probably add a different endpoint mechanism.
